auction-server/src/kernel/entities.rs

```diff
@@ -39,3 +52,41 @@ impl<'de> Deserialize<'de> for PermissionKeySvm {
         Ok(PermissionKeySvm(bytes))
     }
 }
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum ChainType {
+    Evm,
+    Svm,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Evm;
+
+impl Evm {
+    pub fn decode_logs_for_receipt(receipt: &TransactionReceipt) -> Vec<MulticallIssuedFilter> {
+        receipt
+            .logs
+            .clone()
+            .into_iter()
+            .filter_map(|log| MulticallIssuedFilter::decode_log(&log.into()).ok())
+            .collect()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Svm;
+
+impl Svm {
+    pub fn check_tx_size(transaction: &VersionedTransaction) -> Result<(), RestError> {
```

Comment:
maybe move this inside `verification.rs`

Reply:
Isn't it a general function for Svm chains?